Minor fixes

hefloryd opened the pull request:
Allow LIBTELNET_STRICT on libtelnet.c. Fix some warnings resulting from this. Also allow build to complete if Doxygen is not installed.

hefloryd committed:
fix sign compare warnings

hefloryd committed:
enable LIBTELNET_STRICT for libtelnet.c

add_compile_options only affects targets added after the command is
invoked.

hefloryd committed:
fix warning when HAVE_ZLIB undefined

hefloryd committed:
allow build to complete when Doxygen is not found

hefloryd commented:
Enabled C99 to fix C++ style comments warning - perhaps it would be better to just change style to plain C comments instead?

seanmiddleditch reviewed (changes requested):
Mostly looks good. I agree with your last comment that we should just replace the C99 comments with C89 comments instead of requiring C99 to compile, at least for now.

A completely separate PR can do a pass over the code and modernize it to C99. I wouldn't be surprised if that finds and fixes some bugs, too.

hefloryd committed:
remove C++ style comment

C++ style comments are not allowed with -pedantic (unless a standard
is also specified that allows such comments).

hefloryd commented:
Sounds good, replaced C99 commit with new commit changing comment style instead.
